common/OpTestSSH.py

```diff
@@ -155,52 +159,131 @@ def get_console(self):
 
         return self.console
 
-    def run_command(self, command, timeout=60):
-        console = self.get_console()
-        if self.prompt:
-            prompt = self.prompt
+    def set_env(self, console):
+        set_env_list = []
+        console.sendline("exec bash --norc --noprofile")
+        expect_prompt = self.build_prompt() + "$"
+        console.sendline('PS1=' + self.build_prompt())
+        console.expect(expect_prompt)
+        combo_io = (console.before + console.after).lstrip()
+        set_env_list += combo_io.splitlines()
+        # remove the expect prompt since matched generic #
+        del set_env_list[-1]
+        return set_env_list
+
+    def try_sendcontrol(self, console, command):
+        console.sendcontrol('c')
+        try_list = []
+        try_list += self.set_env(console)
+        # we need the env to get the expect_prompt
+        rc = console.expect([expect_prompt, pexpect.TIMEOUT, pexpect.EOF], 10)
```

Comment:
And also expect_prompt is not defined in this function. it fails with below error.
```
======================================================================
ERROR [147.874s]: runTest (testcases.OpTestPNOR.Skiroot)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins_data/sath/pridhivi/debmc/op-test-framework/testcases/OpTestPNOR.py", line 165, in runTest
    self.runTestReadEraseWriteNVRAM()
  File "/home/jenkins_data/sath/pridhivi/debmc/op-test-framework/testcases/OpTestPNOR.py", line 102, in runTestReadEraseWriteNVRAM
    self.pflashErase(nvramInfo['offset'], nvramInfo['length'])
  File "/home/jenkins_data/sath/pridhivi/debmc/op-test-framework/testcases/OpTestPNOR.py", line 58, in pflashErase
    self.c.run_command("pflash -e -f -a %d -s %d" % (offset,length))
  File "/home/jenkins_data/sath/pridhivi/debmc/op-test-framework/common/OpTestSSH.py", line 283, in run_command
    output_list, echo_rc = self.try_sendcontrol(console, command)
  File "/home/jenkins_data/sath/pridhivi/debmc/op-test-framework/common/OpTestSSH.py", line 180, in try_sendcontrol
    rc = console.expect([expect_prompt, pexpect.TIMEOUT, pexpect.EOF], 10)
NameError: global name 'expect_prompt' is not defined

```